scrypto-unit/src/test_runner.rs

```diff
@@ -305,20 +310,38 @@ impl<E: NativeVmExtension, D: TestDatabase> TestRunnerBuilder<E, D> {
             custom_database: database,
             trace: self.trace,
             state_hashing: self.state_hashing,
+            collect_events: self.collect_events,
         }
     }
 
     pub fn build_and_get_epoch(self) -> (TestRunner<E, D>, ActiveValidatorSet) {
+        //---------- Override configs for resource tracker ---------------
+        let bootstrap_trace = false;
+
+        #[cfg(not(feature = "resource_tracker"))]
+        let trace = self.trace;
+        #[cfg(feature = "resource_tracker")]
+        let trace = false;
+
+        let mut collected_events = if self.collect_events {
```

Comment:
Looks like a strange way of setting up `collected_events`. Should just move this line down to line 358